### PR TITLE
#166104085 Admin can view users

### PR DESCRIPTION
**Pull request by Jkadhuwa**
**What does this PR do?**

- Admin can view users

**Description of Task to be completed?**

- Create mocha test for the endpoint

- Create an endpoint for admin to view all users.

**How should this be manually tested?**

- Using postman run hhtp://localhost:8080/api/v1/users

- Then send the request using GET method by pressing the send button

**What are the relevant pivotal tracker stories?**

- #166104085

**Any background context you want to add?**

- NIL

**Screenshots**

- NIL

### CODE REVIEW

File: server/controllers/usersController.js [hunk 1]
```diff
@@ -6,73 +6,66 @@ class UsersController {
...
-				isAdmin
+				isadmin
 			});
 			if (!regUser.createUser()) {
 				res
 					.status(statusCode.CONFLICT)
 					.send({ status: statusCode.CONFLICT, error: 'Email already in use' });
 			}
-			const token = Auth.generateToken(regUser.isadmin, regUser.email);
+			const token = Auth.generateToken(regUser.isadmin, regUser.email, regUser.firstname, regUser.lastname);
```
Comment (meshack-mbuvi) [inline server/controllers/usersController.js]: The token should have `isAdmin` and `email` or just `email` only

File: server/models/userModel.js [hunk 2]
```diff
@@ -5,39 +5,30 @@ class UsersModel {
...
-				user.isAdmin
+				this.payload.firstname,
+				this.payload.lastname,
+				this.payload.email,
+				this.payload.password,
+				this.payload.telephone,
+				this.payload.workaddress,
+				this.payload.nationality,
+				this.payload.status,
+				this.payload.isadmin
```
Comment (meshack-mbuvi) [inline server/models/userModel.js]: You should use array destructuring here like:
```
const {	
    firstname,	
    lastname,	
    email,	
    password,
    telephone,
    workAddress,
    nationality,
    status,
    isAdmin,} = this.payload;
```
Then use it as:
```
const values = [firstname,lastname,...]
```

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

**Review by kodek-sleuth — COMMENTED** [review]
Where are the unit tests for the endpoints

File: server/controllers/usersController.js (hunk 1 above)
Comment (Jkadhuwa) [reply server/controllers/usersController.js]: I use the isAdmin to verify if the user is admin and I planned to use the email in loan endpoints to identify the user.

Comment (meshack-mbuvi) [reply server/controllers/usersController.js]: Did you understand my comments above?

**Comment by Jkadhuwa** [comment]
> Where are the unit tests for the endpoints
@kodek-sleuth  Did you check the files really?

File: server/controllers/usersController.js (hunk 1 above)
Comment (Jkadhuwa) [reply server/controllers/usersController.js]: I get it now. Working on it right away

**Review by meshack-mbuvi — APPROVED** [review]

**Pull request merged**